libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.unique/types.h

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Not attached to this file: The tests used to also exercise the interoperation between `std::unique_lock` and `std::mutex` because that's the only type we used. Now, they don't anymore.

So in theory `std::unique_lock<std::mutex>` could be broken and we wouldn't really notice anymore. Can you add a general-purpose test that is `UNSUPPORTED: no-threads` and that uses `std::unique_lock<std::mutex>` in a couple of different ways. No need to have 100% testing coverage for `unique_lock`'s API, but let's at least make sure that it's not broken since that would be embarrassing.

Reply:
hmm, we still have several tests that still have `no-threads`
- thread.lock.unique.locking/lock.pass.cpp
- thread.lock.unique.cons/mutex.pass.cpp
- thread.lock.unique.cons/mutex_duration.pass.cpp
- thread.lock.unique.cons/mutex_try_to_lock.pass.cpp
- thread.lock.unique.cons/mutex_time_point.pass.cpp

Of these 3 use `std::mutex`: 
- thread.lock.unique.cons/mutex.pass.cpp
- thread.lock.unique.cons/mutex_try_to_lock.pass.cpp
- thread.lock.unique.locking/lock.pass.cpp

I think most of the API surface is tested in those tests. I see couple of  constructors/destructors, lock, unlock, release ... I think maybe we're only missing try_for/try_until, and maybe some of the other assertions on move construction.

In light of those tests still using `std::mutex` do you still want me to add more tests? I'm happy to, but I can't think of much outside the try* APIs that should get exercised.